test/src/tests.js

```diff
@@ -413,4 +416,205 @@ describe('Rokt Forwarder', () => {
+                    'rokt.custom.optimizely.experiment.exp1.variationId',
+                    'var1'
+                );
+            });
+
+            it('should fetch experiment data for multiple experiments', async () => {
+                setupValidOptimizelyMock({
+                    exp1: { id: 'var1' },
+                    exp2: { id: 'var2' },
+                });
+
+                await initAndSelectPlacements({
+                    onboardingExpProvider: 'Optimizely',
+                });
+
+                const attributes =
+                    window.Rokt.selectPlacementsOptions.attributes;
+                attributes.should.have.property(
+                    'rokt.custom.optimizely.experiment.exp1.variationId',
+                    'var1'
+                );
+                attributes.should.have.property(
+                    'rokt.custom.optimizely.experiment.exp2.variationId',
+                    'var2'
+                );
+            });
+        });
+
+        describe('when Optimizely is not properly configured', () => {
+            it('should return empty object when Optimizely is not available', async () => {
+                delete window.optimizely;
+
+                await initAndSelectPlacements({
+                    onboardingExpProvider: 'Optimizely',
+                });
+
+                window.Rokt.selectPlacementsOptions.attributes.should.not.have.property(
+                    'rokt.custom.optimizely'
+                );
+            });
+
+            it('should return empty object when Optimizely state is undefined', async () => {
+                setupInvalidOptimizelyMock(undefined);
+
+                await initAndSelectPlacements({
+                    onboardingExpProvider: 'Optimizely',
+                });
+
+                window.Rokt.selectPlacementsOptions.attributes.should.not.have.property(
+                    'rokt.custom.optimizely'
+                );
+            });
+
+            it('should return empty object when Optimizely state has invalid format', async () => {
+                setupInvalidOptimizelyMock({
+                    someOtherProperty: 'value',
+                    invalidFunction: function () {
+                        return null;
+                    },
+                });
+
+                await initAndSelectPlacements({
+                    onboardingExpProvider: 'Optimizely',
+                });
+
+                window.Rokt.selectPlacementsOptions.attributes.should.not.have.property(
+                    'rokt.custom.optimizely'
+                );
+            });
+
+            it('should return empty object when Optimizely state is missing required methods', async () => {
+                setupInvalidOptimizelyMock({
+                    getVariationMap: function () {
+                        return {};
+                    },
+                    // Missing getActiveExperimentIds
```

Comment:
Nitpick: Updating comment for clarity
```suggestion
                    // Mocking a scenario for when getActiveExperimentIds() method is missing
```